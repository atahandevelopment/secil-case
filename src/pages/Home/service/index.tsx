import api from "../../../instance";

export const GetProducts = async (category: string | undefined, limit: number | undefined) => {
    const queryPage = limit || 30;
    let url = `/products/category/${category}?limit=${queryPage}`;
    if (!category) url = `/products?limit=${queryPage}`;
    return await api.get(url);
}