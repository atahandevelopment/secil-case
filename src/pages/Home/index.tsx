/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { useEffect, useState } from "react";
import Card from "../../components/Card";
import DefaultLayout from "../../components/Layout";
import { GetProducts } from "./service";
import { ProductTypes } from "./types";
import "./styles/home.css";
import { Link, useParams } from "react-router-dom";

const Home = () => {
  const [products, setProducts] = useState<ProductTypes[]>([]);
  const { category } = useParams();
  const categoryQuery = category || undefined;
  // Sayfa ilk render edildiğinde parametreyi alıp query'yi gelen parametreye göre düzenliyorum ve isteği category parametresine göre yapıyorum.
  useEffect(() => {
    GetProducts(categoryQuery, 30)
      .then((res) => {
        setProducts(res.data);
      })
      .catch((err) => console.error(err));
  }, [categoryQuery]);

  if (products.length === 0) {
    return <div className="w-screen h-screen flex justify-center items-center">
    <div className="loader"></div>
  </div>;
  }

  return (
    <DefaultLayout>
      <div data-testid="home-component" className="list-container">
        <div className="products-list">
          {products && products.length > 0 ? (
            products.map((product, index) => (
              <Link key={index} to={`/product-detail/${product?.id}`}>
                <Card data={product} />
              </Link>
            ))
          ) : (
            <></>
          )}
        </div>
      </div>
    </DefaultLayout>
  );
};

export default Home;
