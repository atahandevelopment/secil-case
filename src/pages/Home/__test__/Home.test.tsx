import { expect, it } from 'vitest';
import Home from '..';
import renderer from 'react-test-renderer';

it('renders Home component correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
