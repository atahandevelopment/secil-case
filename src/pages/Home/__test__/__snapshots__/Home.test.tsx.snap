// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Home component correctly 1`] = `
<div
  className="w-screen h-screen flex justify-center items-center"
>
  <div
    className="loader"
  />
</div>
`;
